operator/internal/manifests/env_vars.go

```diff
@@ -0,0 +1,85 @@
+package manifests
+
+import (
+	"strings"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/imdario/mergo"
+	"github.com/operator-framework/operator-lib/proxy"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	httpProxyKey  = "HTTP_PROXY"
+	httpsProxyKey = "HTTPS_PROXY"
+	noProxyKey    = "NO_PROXY"
+)
+
+func configureProxyEnv(spec *lokiv1.ClusterProxy, pod *corev1.PodSpec) error {
+	if spec == nil {
+		return nil
+	}
+
+	for _, envVar := range proxy.ProxyEnvNames {
+		resetProxyVar(pod, envVar)
```

Comment:
For now this is the only place we set environment variables. We can't use `WithOvveride` here because it would replace the entire env vars slice and we may introduce somewhere else env vars. This would make debugging hard for no reason. 

In general I suggested to put here a reset because double entries spam the kube audit log. I imagine you can introduce env vars on the pod controller level. So this is just for safety.